servant-server/src/Servant/Server/Internal.hs

```diff
@@ -880,5 +882,29 @@ type HasServerArrowTypeError a b =
     ':$$: 'ShowType b
 #endif
 
+-- | Ignore @'Fragment'@ in server handlers.
+-- See <https://ietf.org/rfc/rfc2616.html#section-15.1.3> for more details.
+--
+-- Example:
+--
+-- > type MyApi = "books" :> Fragment Text :> Get '[JSON] [Book]
+-- >
+-- > server :: Server MyApi
+-- > server = getBooksBy
+-- >   where getBooksBy :: Handler [Book]
+-- >         getBooksBy = ...return all books...
+#ifdef HAS_TYPE_ERROR
+instance (OnlyOneFragment api, HasServer api context, FromHttpApiData a1)
+#else
+instance (HasServer api context, FromHttpApiData a1)
+#endif
+    => HasServer (Fragment a1 :> api) context where
+  type ServerT (Fragment a1 :> api) m = ServerT api m
+
+  route Proxy context subserver = route (Proxy :: Proxy api) context subserver 
```

Comment:
You can eta-reduce to have `route _ = route (Proxy :: Proxy api)`, similarly for `hoist`.

Reply:
Fixed.